tests/test_s3_to_json.py

```diff
@@ -83,117 +83,309 @@ def json_file_basenames_dict(self):
+                }
+        ]
         sample_metadata = {
-            "Metadata": {
-                "type": "SymptomLog",
+                "type": "GarminDailySummary",
                 "start_date": datetime.datetime(2022, 1, 12, 0, 0),
-                "end_date": datetime.datetime(2023, 1, 14, 0, 0)
-            }
+                "end_date": datetime.datetime(2023, 1, 14, 0, 0),
         }
-        workflow_run_properties = {
-            "namespace": namespace,
-            "json_prefix": "raw-json",
-            "json_bucket": "json-bucket",
+        transformed_json = s3_to_json.transform_json(
+                json_obj={"TimeOffsetHeartRateSamples": time_offset_heartrate_samples},
+                dataset_identifier=sample_metadata["type"],
+                metadata=sample_metadata
+        )
+
+        assert (
+            sample_metadata["start_date"].isoformat()
+            == transformed_json["export_start_date"]
+        )
+        assert (
+            sample_metadata["end_date"].isoformat()
+            == transformed_json["export_end_date"]
+        )
+        assert all(
+                [
+                    obj in transformed_json["TimeOffsetHeartRateSamples"]
+                    for obj in transformed_time_offset_heartrate_samples
+                ]
+        )
+
+    def test_transform_json_garmin_two_levels_down(self):
+        epoch_summaries = {
+                "0": 60.0,
+                "1": 61.0,
+                "2": 99.0
         }
-        with zipfile.ZipFile(io.BytesIO(s3_obj["Body"])) as z:
-            output_file = s3_to_json.write_file_to_json_dataset(
-                z=z,
-                json_path="SymptomLog_20230112-20230114.json",
-                dataset_identifier="SymptomLog",
-                metadata=sample_metadata["Metadata"],
-                workflow_run_properties=workflow_run_properties,
-                delete_upon_successful_upload=False,
-            )
+        transformed_epoch_summaries = [
+                {
+                    "OffsetInSeconds": 0,
+                    "Value": 60.0
+                },
+                {
+                    "OffsetInSeconds": 1,
+                    "Value": 61.0
+                },
+                {
+                    "OffsetInSeconds":2,
+                    "Value": 99.0
+                }
+        ]
+        sample_metadata = {
+                "type": "GarminHealthSnapshotSummary",
+                "start_date": datetime.datetime(2022, 1, 12, 0, 0),
+                "end_date": datetime.datetime(2023, 1, 14, 0, 0),
+        }
+        transformed_json = s3_to_json.transform_json(
+                json_obj={
+                    "Summaries": [
+                        {
+                            "EpochSummaries": epoch_summaries,
+                            "Dummy": 1
+                        },
+                        {
+                            "EpochSummaries": epoch_summaries,
+                            "Dummy": 1
```

Comment:
Should this dummy value be different?

Reply:
That's not really necessary, but I've removed `Dummy` from the second object anyways since we don't use it as part of an assert statement.